front/src/modules/rollingStock/helpers/utils.ts

```diff
@@ -202,9 +206,23 @@ export const checkRollingStockFormValidity = (
       ...pick(RS_REQUIRED_FIELDS, invalidFields),
       ...omit(rollingStockForm, invalidFields),
     } as RollingStockParametersValidValues,
+    invalidEffortCurves: Object.values(effortCurves || {}).some(({ curves }) =>
+      curves.some(({ curve }) => isInvalidCurve(curve))
+    ),
   };
 };
 
+export const isInvalidCurve = (curve: EffortCurveForm): boolean => {
+  let isInvalid = curve.max_efforts.some((value) => value !== undefined && value < 0);
+  if (!isInvalid) {
+    const uniqueValues = new Set(curve.speeds);
+    const hasDuplicates = uniqueValues.size !== curve.speeds.length;
+    const someNegativeValue = curve.speeds.some((value) => value !== undefined && value < 0);
+    isInvalid = hasDuplicates || someNegativeValue;
+  }
+  return isInvalid;
+};
```

Comment:
As we can't enter any negative values for speeds or efforts, you could refacto this function like that : 
```suggestion
export const isInvalidCurve = (curve: EffortCurveForm): boolean => {
  const uniqueValues = new Set(curve.speeds);
  return uniqueValues.size !== curve.speeds.length;
};
```